src/lib/util/Config.ts

```diff
@@ -79,6 +90,31 @@ export class Config {
 		return this.configContent.engines.find(e => e.name === name);
 	}
 
+	public getOverriddenConfigPath(name: string): string {
+		const defaultValue = '';
+		const engineConfig = this.getEngineConfig(name);
+
+		if (!this.shouldUseDefaultConfig(name)) {
+			if (!engineConfig || !engineConfig.overriddenConfigPath) {
+				throw new SfdxError(`Please set "overriddenConfigPath" with config path to override for engine ${name} in ${CONFIG_FILE_PATH}. If you want to use default value, please set "useDefaultConfig" to true.`);
+			}
+			return engineConfig.overriddenConfigPath;
+		}
+		return defaultValue;
+	}
+
+	private shouldUseDefaultConfig(name: string): boolean {
+		const engineConfig = this.getEngineConfig(name);
+		const defaultValue = true;
```

Comment:
The error message above states "If you want to use default value, please set "useDefaultConfig" to true.`", which made me thing this would default to false if useDefaultConfig wasn't present?

Reply:
You are absolutely right. Great catch.

Reply:
Yeah, the check is confusing. A user could leave a overridden path value and still choose not to use it. Let me fix this.